server/src/models/people/models.js

```diff
@@ -1,69 +1,66 @@
-import { camel_case_headcount_model_names } from "./headcount_model_utils.js";
+import _ from "lodash";
+import mongoose from "mongoose";
 
-export default function define_resource_models(model_singleton) {
-  const _headcount_records_by_model_and_dept_code = {};
+import {
+  create_resource_by_id_attr_dataloader,
+  create_resource_by_foreignkey_attr_dataloader,
+} from "../loader_utils.js";
+import {
+  pkey_type,
+  parent_fkey_type,
+  fyear_type,
+  number_type,
+  str_type,
+  bilingual,
+} from "../model_utils.js";
 
-  class HeadcountModel {
-    constructor(obj) {
-      Object.assign(this, obj);
-    }
-    static model_name() {
-      throw new Error(`
-        Error: bad OOP alert, don't use HeadcountModels class directly.
-        Need to create a new class extending it that overwrites the static 
-        model_name() function with one that returns the name of the actual model.
-      `);
-    }
-    static get_model_name() {
-      try {
-        return this.model_name();
-      } catch (e) {
-        console.log(e);
-      }
-    }
-    static register(obj) {
-      const { dept_code } = obj;
+export default function (model_singleton) {
+  const EmployeeWholeDataSchema = mongoose.Schema({
+    dept_code: pkey_type(),
+    data: [
+      {
+        dimension: str_type,
+        by_year: [
+          {
+            year: fyear_type(),
+            value: number_type,
+          },
+        ],
+        avg_share: number_type,
+      },
+    ],
+  });
 
-      const inst = new HeadcountModel(obj);
+  const EmployeeAveragesSchema = mongoose.Schema({
```

Comment:
Small name nitpick, since this is specifically just average age right now (and we don't specifically plan for future average datasets), I'd just specifically name it `EmployeeAverageAgeSchema`.